edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ModelIdStore.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core
+{
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class ModelIdStore : IModelIdStore
+    {
+        readonly IKeyValueStore<string, string> modelIdEntityStore;
+
+        public ModelIdStore(IKeyValueStore<string, string> modelIdEntityStore)
+        {
+            this.modelIdEntityStore = Preconditions.CheckNotNull(modelIdEntityStore, nameof(modelIdEntityStore));
+        }
+
+        public Task SetModelId(string id, string modelId)
+        {
+            Preconditions.CheckNonWhiteSpace(id, nameof(id));
+            return !string.IsNullOrWhiteSpace(modelId) ? this.modelIdEntityStore.Put(id, modelId) : Task.CompletedTask;
+        }
+
+        public async Task<string> GetModelId(string id)
+        {
+            Preconditions.CheckNonWhiteSpace(id, nameof(id));
+            Option<string> modelId = await this.modelIdEntityStore.Get(id);
+            return modelId.GetOrElse(string.Empty);
```

Comment:
>return modelId.GetOrElse(string.Empty); [](start = 12, length = 39)

seems like you're returning potentially a null-string here, only to have the callers convert it back to an Option<string> again.  Why not just return an Option<string> from this interface?

Reply:
Fair point! Changed